modules/admin-ui-frontend/app/scripts/shared/partials/modals/embedding-code.html

```diff
@@ -1,57 +1,112 @@
 <section ng-show="open" ng-keyup="keyUp($event)" tabindex="1" class="modal modal-animation ng-hide" id="embedding-code">
   <header>
     <a class="fa fa-times close-modal" ng-click="close()"></a>
-    <h2 translate="CONFIRMATIONS.ACTIONS.SHOW.EMBEDDING_CODE"><!-- Confirm --></h2>
+    <h2 translate="CONFIRMATIONS.ACTIONS.SHOW.EMBEDDING_CODE">
+      <!-- Confirm -->
+    </h2>
   </header>
 
   <script>
     var engageUrl = "<SERVER_URL>";
+    var downloadLinks = [];
 
-    $(document).ready(function() {
+    $(document).ready(function () {
       var reply = $.ajax({
         url: "/api/info/organization/properties/engageuiurl",
         timeout: 3000,
-        success: function(result) {
+        success: function (result) {
           engageUrl = result['org.opencastproject.engage.ui.url'];
           if (typeof engageUrl === "undefined") {
             engageUrl = "<SERVER_URL>";
           }
-        }
-      });
-    });
 
-    var embs = $(".embedSizeButton");
-    for (var i = 0; i < embs.length; i = i + 1) {
-      var e = embs[i];
-      e.onclick = function(event) {
-        var value = event.target ? event.target.textContent : event.toElement.textContent;
-        if (value) {
-          var embedSizeButtons = $(".embedSizeButton");
-          for (var j = 0; j < embedSizeButtons.length; j++) {
-            if (embedSizeButtons[j].id == value) {
-              embedSizeButtons[j].classList.add("embedSizeButtonSelected");
-            } else {
-              embedSizeButtons[j].classList.remove("embedSizeButtonSelected");
+          //getting all available formats for direct download
+          var apiCall = engageUrl + "/api/events/" + $("#id_video").html() + "?withpublications=true";
+          var reply = $.ajax({
+            url: apiCall,
+            timeout: 3000,
+            success: function (result) {
+              var pubs = result["publications"];
+              for (var i = 0; i < pubs.length; i++) {
+                if (pubs[i]["channel"] == "engage-player") {
+                  var mediaCnt = pubs[i]["media"].length;
+                  var media = pubs[i]["media"];
+                  media.sort((a, b) => (a["width"] > b["width"]) ? -1 : ((b["width"] > a["width"]) ? 1 : 0));
+
+                  //add all necessary divs
+                  for (var j = 0; j < mediaCnt; j++) {
```

Comment:
This is simpler and better to read:

```suggestion
                  var media = pubs[i]["media"];
                  media.sort((a, b) => b.width - a.width);

                  // add all necessary divs
                  for (var j = 0; j < media.length; j++) {
```